src/test/rgw/sfs/test_rgw_sfs_sqlite_versioned_objects.cc

```diff
@@ -0,0 +1,421 @@
+  auto obj3 = createTestVersionedObject(3, TEST_OBJECT_ID, "3");
+  db_versioned_objects->storeVersionedObject(obj3);
+
+  db_versioned_objects->removeVersionedObject(obj2.id);
+  auto object_ids = db_versioned_objects->getVersionedObjectIDs();
+  EXPECT_EQ(object_ids.size(), 2);
+
+  EXPECT_EQ(1, object_ids[0]);
+  EXPECT_EQ(3, object_ids[1]);
+
+  auto ret_object = db_versioned_objects->getVersionedObject(obj2.id);
+  ASSERT_FALSE(ret_object.has_value());
+}
+
+TEST_F(TestSFSSQLiteVersionedObjects, RemoveObjectThatDoesNotExist) {
+  auto ceph_context = std::make_shared<CephContext>(CEPH_ENTITY_TYPE_CLIENT);
+  ceph_context->_conf.set_val("rgw_sfs_data_path", getTestDir());
+
+  EXPECT_FALSE(fs::exists(getDBFullPath()));
+
+  // Create the object, we need if because of foreign key constrains
+  createObject(TEST_USERNAME, TEST_BUCKET, TEST_OBJECT_ID, ceph_context.get());
+
+  auto db_versioned_objects = std::make_shared<SQLiteVersionedObjects>(ceph_context.get());
+
+  auto obj1 = createTestVersionedObject(1, TEST_OBJECT_ID, "1");
+  db_versioned_objects->storeVersionedObject(obj1);
+  auto obj2 = createTestVersionedObject(2, TEST_OBJECT_ID, "2");
+  db_versioned_objects->storeVersionedObject(obj2);
+  auto obj3 = createTestVersionedObject(3, TEST_OBJECT_ID, "3");
+  db_versioned_objects->storeVersionedObject(obj3);
+
+  db_versioned_objects->removeVersionedObject(10);
+  auto object_ids = db_versioned_objects->getVersionedObjectIDs();
+  EXPECT_EQ(object_ids.size(), 3);
+
+  EXPECT_EQ(1, object_ids[0]);
+  EXPECT_EQ(2, object_ids[1]);
+  EXPECT_EQ(2, object_ids[1]);
+
+  auto ret_object = db_versioned_objects->getVersionedObject(obj2.id);
+  ASSERT_TRUE(ret_object.has_value());
+}
+
+TEST_F(TestSFSSQLiteVersionedObjects, CreateAndUpdate) {
+  auto ceph_context = std::make_shared<CephContext>(CEPH_ENTITY_TYPE_CLIENT);
+  ceph_context->_conf.set_val("rgw_sfs_data_path", getTestDir());
+
+  EXPECT_FALSE(fs::exists(getDBFullPath()));
+  auto db_versioned_objects = std::make_shared<SQLiteVersionedObjects>(ceph_context.get());
+
+  // Create the object, we need if because of foreign key constrains
+  createObject(TEST_USERNAME, TEST_BUCKET, TEST_OBJECT_ID, ceph_context.get());
+
+  auto versioned_object = createTestVersionedObject(1, TEST_OBJECT_ID, "1");
+
+  db_versioned_objects->storeVersionedObject(versioned_object);
+  EXPECT_TRUE(fs::exists(getDBFullPath()));
+
+  auto ret_ver_object = db_versioned_objects->getVersionedObject(versioned_object.id);
+  ASSERT_TRUE(ret_ver_object.has_value());
+  compareVersionedObjects(versioned_object, *ret_ver_object);
+
+  // update the size
+  versioned_object.size = 1999;
+  db_versioned_objects->storeVersionedObject(versioned_object);
+
+  ret_ver_object = db_versioned_objects->getVersionedObject(versioned_object.id);
+  ASSERT_TRUE(ret_ver_object.has_value());
+  ASSERT_EQ(1999, ret_ver_object->size);
+  compareVersionedObjects(versioned_object, *ret_ver_object);
+}
```

Comment:
shouldn't we be checking that there exists more than one entry for this object?

Reply:
I've added that